CRM/CiviAwards/Service/AwardCaseFilterPreProcess.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * Award case filter pre-process class.
+ */
+class CRM_CiviAwards_Service_AwardCaseFilterPreProcess {
+
+  /**
+   * Function that handles pre-processing for a case details relating to awards.
+   *
+   * @param array $requestParams
+   *   API request parameters.
+   */
+  public function onCreate(array &$requestParams) {
```

Comment:
This name actually depicts the event on which this is fired.  The event on which this is called is not related to the creation of an API. Maybe we can rename to `onCaseDetailsGet` or better still `alterRequestParams` to depict what the function is doing.